common/sagas/wallet/wallet.ts

```
@@ -0,0 +1,250 @@
+  getKeystoreWallet,
+  Web3Wallet,
+  WalletConfig
+} from 'libs/wallet';
+import { NODES, initWeb3Node } from 'config/data';
+import { SagaIterator } from 'redux-saga';
+import { apply, call, fork, put, select, takeEvery, take } from 'redux-saga/effects';
+import { getNodeLib } from 'selectors/config';
+import { getTokens, getWalletInst, getWalletConfig } from 'selectors/wallet';
+import { getCustomTokens } from 'selectors/customTokens';
+import translate from 'translations';
+import Web3Node, { isWeb3Node } from 'libs/nodes/web3';
+import { loadWalletConfig, saveWalletConfig } from 'utils/localStorage';
+import { getTokenBalances } from './helpers';
+
+export function* updateAccountBalance(): SagaIterator {
+  try {
+    yield put(setBalancePending());
+    const wallet: null | IWallet = yield select(getWalletInst);
+    if (!wallet) {
+      return;
+    }
+    const node: INode = yield select(getNodeLib);
+    const address = yield apply(wallet, wallet.getAddressString);
+    // network request
+    const balance: Wei = yield apply(node, node.getBalance, [address]);
+    yield put(setBalanceFullfilled(balance));
+  } catch (error) {
+    yield put(setBalanceRejected());
+  }
+}
+
+export function* updateTokenBalances(): SagaIterator {
+  try {
+    const wallet: null | IWallet = yield select(getWalletInst);
+    const config: null | WalletConfig = yield select(getWalletConfig);
+    if (!wallet || !config || !config.tokens) {
+      return;
+    }
+
+    // Get the token objects for the tokens in the config
+    const allTokens = yield select(getTokens);
+    const tokens = config.tokens
+      .map(symbol => allTokens.find(t => t.symbol === symbol))
+      .filter(token => !!token);
+
+    // Fetch & Set
+    yield put(setTokenBalancesPending());
+    const tokenBalances = yield call(getTokenBalances, wallet, tokens);
+    yield put(setTokenBalancesFulfilled(tokenBalances));
+  } catch (error) {
+    console.error('Failed to get token balances', error);
```

Log

---

If I use `info`, `warn`, or `error`, it's intentionally left in. Helps for debugging instead of just swallowing the error.

---

Understood